src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const tempObject = Object.assign({}, state);
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(tempObject, obj.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const el of obj.keysToRemove) {
+          for (const key in tempObject) {
```

Comment:
why do you need this loop? you can just check if property is in object with .hasOwnProperty method

Reply:
Fixed